### PR TITLE
304 no storage panic backport

### PR DESCRIPTION
**Pull request by andrewwiik**
Fixes: #3273

Commit Cherry Picked: d3b6409ea3f52851aa63fde8120af5cfc430db8b

### CODE REVIEW

File: bin/varnishd/cache/cache_fetch.c [hunk 1]
```diff
@@ -716,7 +716,12 @@ vbf_stp_condfetch(struct worker *wrk, struct busyobj *bo)
 	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
 	CHECK_OBJ_NOTNULL(bo, BUSYOBJ_MAGIC);
 
-	AZ(vbf_beresp2obj(bo));
+	if (vbf_beresp2obj(bo)) {
+		(void)VFP_Error(bo->vfc, "Could not get storage in vbf_stp_condfetch");
+		vbf_cleanup(bo);
```
Comment (rezan) [inline bin/varnishd/cache/cache_fetch.c:721]: I think `vbf_cleanup()` is missing?

Comment (rezan) [reply bin/varnishd/cache/cache_fetch.c:721]: I think it just needs to mirror the failure in a normal fetch:

https://github.com/varnishcache/varnish-cache/blob/29a1a8243dbef3d973aec28dc90403188c1dc8e7/bin/varnishd/cache/cache_fetch.c#L604-L610

Comment (nigoroll) [reply bin/varnishd/cache/cache_fetch.c:721]: @rezan I asked myself the same questions, and the answer is here: https://github.com/varnishcache/varnish-cache/pull/3274#issuecomment-608563773

Comment (andrewwiik) [reply bin/varnishd/cache/cache_fetch.c:721]: Yes @nigoroll is correct, I had fixed this but guess I reverted something and didn't fix it twice.

Comment (rezan) [reply bin/varnishd/cache/cache_fetch.c:721]: I did the load test with the code from my previous comment and everything worked as described.

**Comment by andrewwiik** [comment]
Closing in favor of #3326

**Pull request closed without merging**